dual_panda_effort_joint_trajectory_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - rarm_joint1
    - rarm_joint2
    - rarm_joint3
    - rarm_joint4
    - rarm_joint5
    - rarm_joint6
    - rarm_joint7
    - larm_joint1
    - larm_joint2
    - larm_joint3
    - larm_joint4
    - larm_joint5
    - larm_joint6
    - larm_joint7
  constraints:
    goal_time: 0.5
    larm_joint1:
      goal: 0.05
    larm_joint2:
      goal: 0.05
    larm_joint3:
      goal: 0.05
    larm_joint4:
      goal: 0.05
    larm_joint5:
      goal: 0.05
    larm_joint6:
      goal: 0.05
    larm_joint7:
      goal: 0.05
    rarm_joint1:
      goal: 0.05
    rarm_joint2:
      goal: 0.05
    rarm_joint3:
      goal: 0.05
    rarm_joint4:
      goal: 0.05
    rarm_joint5:
      goal: 0.05
    rarm_joint6:
      goal: 0.05
    rarm_joint7:
      goal: 0.05
  gains:
    rarm_joint1: {p: 600, d: 30, i: 0, i_clamp: 1}
    rarm_joint2: {p: 600, d: 30, i: 0, i_clamp: 1}
    rarm_joint3: {p: 600, d: 30, i: 0, i_clamp: 1}
    rarm_joint4: {p: 600, d: 30, i: 0, i_clamp: 1}
    rarm_joint5: {p: 250, d: 10, i: 0, i_clamp: 1}
    rarm_joint6: {p: 150, d: 10, i: 0, i_clamp: 1}
    rarm_joint7: {p: 50, d: 5, i: 0, i_clamp: 1}
    larm_joint1: {p: 600, d: 30, i: 0, i_clamp: 1}
    larm_joint2: {p: 600, d: 30, i: 0, i_clamp: 1}
    larm_joint3: {p: 600, d: 30, i: 0, i_clamp: 1}
    larm_joint4: {p: 600, d: 30, i: 0, i_clamp: 1}
    larm_joint5: {p: 250, d: 10, i: 0, i_clamp: 1}
    larm_joint6: {p: 150, d: 10, i: 0, i_clamp: 1}
    larm_joint7: {p: 50, d: 5, i: 0, i_clamp: 1}

dual_panda_pose_cartesian_controller:
  type: franka_example_controllers/CartesianPoseExampleController
  joints:
    - rarm_joint1
    - rarm_joint2
    - rarm_joint3
    - rarm_joint4
    - rarm_joint5
    - rarm_joint6
    - rarm_joint7
    - larm_joint1
    - larm_joint2
    - larm_joint3
    - larm_joint4
    - larm_joint5
    - larm_joint6
    - larm_joint7
  constraints:
    goal_time: 0.5
    larm_joint1:
      goal: 0.05
    larm_joint2:
      goal: 0.05
    larm_joint3:
      goal: 0.05
    larm_joint4:
      goal: 0.05
    larm_joint5:
      goal: 0.05
    larm_joint6:
      goal: 0.05
    larm_joint7:
      goal: 0.05
    rarm_joint1:
      goal: 0.05
    rarm_joint2:
      goal: 0.05
    rarm_joint3:
      goal: 0.05
    rarm_joint4:
      goal: 0.05
    rarm_joint5:
      goal: 0.05
    rarm_joint6:
      goal: 0.05
    rarm_joint7:
      goal: 0.05
  gains:
    rarm_joint1: {p: 600, d: 30, i: 0, i_clamp: 1}
    rarm_joint2: {p: 600, d: 30, i: 0, i_clamp: 1}
    rarm_joint3: {p: 600, d: 30, i: 0, i_clamp: 1}
    rarm_joint4: {p: 600, d: 30, i: 0, i_clamp: 1}
    rarm_joint5: {p: 250, d: 10, i: 0, i_clamp: 1}
    rarm_joint6: {p: 150, d: 10, i: 0, i_clamp: 1}
    rarm_joint7: {p: 50, d: 5, i: 0, i_clamp: 1}
    larm_joint1: {p: 600, d: 30, i: 0, i_clamp: 1}
    larm_joint2: {p: 600, d: 30, i: 0, i_clamp: 1}
    larm_joint3: {p: 600, d: 30, i: 0, i_clamp: 1}
    larm_joint4: {p: 600, d: 30, i: 0, i_clamp: 1}
    larm_joint5: {p: 250, d: 10, i: 0, i_clamp: 1}
    larm_joint6: {p: 150, d: 10, i: 0, i_clamp: 1}
    larm_joint7: {p: 50, d: 5, i: 0, i_clamp: 1}



dual_arm_cartesian_impedance_example_controller:
    type: franka_example_controllers/DualArmCartesianImpedanceExampleController
    right:
        arm_id: rarm
        joint_names:
            - rarm_joint1
            - rarm_joint2
            - rarm_joint3
            - rarm_joint4
            - rarm_joint5
            - rarm_joint6
            - rarm_joint7
    left:
        arm_id: larm
        joint_names:
            - larm_joint1
            - larm_joint2
            - larm_joint3
            - larm_joint4
            - larm_joint5
            - larm_joint6
            - larm_joint7

    
rarm_state_controller:
  type: franka_control/FrankaStateController
  arm_id: rarm
  joint_names:
    - rarm_joint1
    - rarm_joint2
    - rarm_joint3
    - rarm_joint4
    - rarm_joint5
    - rarm_joint6
    - rarm_joint7
  publish_rate: 30  # [Hz]

larm_state_controller:
  type: franka_control/FrankaStateController
  arm_id: larm
  joint_names:
    - larm_joint1
    - larm_joint2
    - larm_joint3
    - larm_joint4
    - larm_joint5
    - larm_joint6
    - larm_joint7
  publish_rate: 30  # [Hz]

dual_arm_cartesian_pose_controller:
  type: franka_example_controllers/DualArmCartesianPoseController
  left:
    arm_id: larm
    joint_names:
      - larm_joint1
      - larm_joint2
      - larm_joint3
      - larm_joint4
      - larm_joint5
      - larm_joint6
      - larm_joint7
  right:
    arm_id: rarm
    joint_names:
      - rarm_joint1
      - rarm_joint2
      - rarm_joint3
      - rarm_joint4
      - rarm_joint5
      - rarm_joint6
      - rarm_joint7
            